src/geometry.F90

```diff
@@ -1333,24 +1314,19 @@ subroutine distance_to_boundary(p, dist, surface_crossed, lattice_translation)
           dist = d_surf
           surface_crossed = level_surf_cross
           lattice_translation(:) = [0, 0, 0]
-          final_coord => coord
+          next_level = j
         end if
       else
         if ((dist - d_lat)/dist >= FP_REL_PRECISION) then
           dist = d_lat
           surface_crossed = None
           lattice_translation(:) = level_lat_trans
-          final_coord => coord
+          next_level = j
         end if
       end if
 
-      coord => coord % next
-
     end do LEVEL_LOOP
 
-    ! Move particle to appropriate coordinate level
-    if (associated(final_coord)) p % coord => final_coord
```

Comment:
Removing this line is essentially why test_filter_universe changed. Since the universe filter looked for `p % coord % universe` before, if distance_to_boundary caused the coordinate level to change (even though the particle hasn't really "left" that level yet), it would result in a different universe being scored to. Now, since the check is on `p % coord(p % n_coord) % universe`, the apparent universe won't change until the next call to find_cell, which doesn't happen until after tracklength tallies are scored... I know this is a little confusing, but I do think the behavior now is more correct, or at least consistent.